code/game/objects/structures/crates_lockers/closets.dm

```diff
@@ -109,6 +123,46 @@
 	. += emissive_appearance(icon, "locked", alpha = src.alpha)
 	. += locked ? "locked" : "unlocked"
 
+/// Animates the closet door opening and closing
+/obj/structure/closet/proc/animate_door(closing = FALSE)
+	if(door_anim_time)
+		if(!door_obj)
+			door_obj = new
+		vis_contents += door_obj
+		door_obj.icon = icon
+		door_obj.icon_state = "[icon_door || icon_state]_door"
+		is_animating_door = TRUE
+		var/num_steps = door_anim_time / world.tick_lag
+		for(var/step in 0 to num_steps)
+			var/angle = door_anim_angle * (closing ? 1 - (step/num_steps) : (step/num_steps))
+			var/matrix/door_transform = get_door_transform(angle)
+			var/door_state = angle >= 90 ? "[icon_door_override ? icon_door : icon_state]_back" : "[icon_door || icon_state]_door"
+			var/door_layer = angle >= 90 ? FLOAT_LAYER : ABOVE_MOB_LAYER
+
+			if(step == 0)
+				door_obj.transform = door_transform
+				door_obj.icon_state = door_state
+				door_obj.layer = door_layer
```

Comment:
I'd like it if this was pulled out of the for loop, and just done at the start

Reply:
I see what you're saying, but I'm not sure how I would do that without making a visual mess of the for loop.

Reply:
ah it'd mean duplicating the icon and layer checks. ahhhhhh pain.
Just leave it